database/ffldb/db.go

```diff
@@ -2001,6 +2001,28 @@ func (db *db) Close() error {
 	return closeErr
 }
 
+// Flush writes all outstanding cached entries to disk.
+//
+// This function is part of the database.DB interface implementation.
+func (db *db) Flush() error {
+	// Since all transactions have a read lock on this mutex, this will
+	// cause Flush to wait for all readers to complete.
+	db.closeLock.Lock()
+	defer db.closeLock.Unlock()
+
+	if db.closed {
+		return makeDbErr(database.ErrDbNotOpen, errDbNotOpenStr)
+	}
+
+	// NOTE: Since the above lock waits for all transactions to finish and
+	// prevents any new ones from being started, it is safe to flush the
+	// cache and clear all state without the individual locks.
```

Comment:
I feel like this should be clear all state**s** but not sure at all.

Reply:
Singular is fine, imo.  It's referring to the entire overall state of the cache.

Reply:
Ok.